addons/mail/static/src/models/welcome_view/welcome_view.js

```diff
@@ -120,20 +115,32 @@ function factory(dependencies) {
 
     WelcomeView.fields = {
         /**
-         * The channel to redirect to once the user clicks on the 'joinButton'.
+         * States the channel to redirect to once the user clicks on the
+         * 'joinButton'.
          */
         channel: one2one('mail.thread', {
+            readonly: true,
             required: true,
         }),
         /**
-         * Used as a value for `id`, `for`, and `name` attributes of the guest
-         * name input and its label. Necessary to ensure the uniqueness.
+         * States the welcome view to be displayed.
+         */
+        discussPublicRoot: one2one('mail.discuss_public_root', {
+            inverse: 'welcomeView',
+            readonly: true,
+        }),
+        /**
+         * States the value to use for `id`, `for`, and `name` attributes of
+         * the guest name input and its label. Necessary to ensure the
+         * uniqueness.
          */
         guestNameInputUniqueId: attr({
             default: `o_WelcomeView_guestNameInput_${getNextGuestNameInputId()}`,
```

Comment:
I think this gives always the same value as this will call the function when the code is parsed.